### PR TITLE
Allow specifying pthread CPU affinity

### PR DESCRIPTION
**Pull request by ferdymercury**
# This Pull request:

## Changes or fixes:
Allows associating a TThread run with a specific CPU on Linux

## Checklist:

- [x] tested changes locally
- [x] updated the docs (if necessary)

### CODE REVIEW

**Comment by bellenot** [comment]
@phsft-bot build

**Review by pcanal — APPROVED** [review]

File: core/thread/src/TPosixThread.cxx [hunk 1]
```diff
@@ -28,13 +28,20 @@ ClassImp(TPosixThread);
...
    int det;
    pthread_t id;
    pthread_attr_t *attr = new pthread_attr_t;
 
    pthread_attr_init(attr);
+   
+   if (affinity >= 0) {
+      cpu_set_t cpuset;
+      CPU_ZERO(&cpuset);
+      CPU_SET(affinity,&cpuset);
```
Comment (pcanal) [inline core/thread/src/TPosixThread.cxx:42]:
```suggestion
      CPU_SET(affinity, &cpuset);
```

**Comment by Axel-Naumann** [comment]
@pcanal is it worth coloring a limping horse? (That must be a proverb somewhere.) I.e. do we still want to grow `TThread`'s feature space?

**Comment by pcanal** [comment]
bah ... the coloring is done ...

**Comment by Axel-Naumann** [comment]
@pcanal you overlooked a compilation error!

@ferdymercury I will have to revert this; could you create a new PR that works also on macOS?

**Comment by ferdymercury** [comment]
sorry, my bad. It looks as it is only specific to linux. I will create a OS guard for the moment.

**Comment by pcanal** [comment]
> @pcanal you overlooked a compilation error!

yep, I am not sure how that happened :( ... Thanks for fixing it!

**Pull request merged**